libbindgen/src/ir/comp.rs

```diff
@@ -236,6 +236,8 @@ pub struct CompInfo {
     /// Used to detect if we've run in a has_destructor cycle while cycling
     /// around the template arguments.
     detect_has_destructor_cycle: Cell<bool>,
+
+    is_declaration: bool,
```

Comment:
let's rename this and the method to `is_forward_declaration`, and also add a doc comment here describing what this indicates.